controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/UpdateStreamTask.java

```diff
@@ -42,16 +48,61 @@ public UpdateStreamTask(final StreamMetadataTasks streamMetadataTasks,
 
     @Override
     public CompletableFuture<Void> execute(final UpdateStreamEvent request) {
-        CompletableFuture<Void> result = new CompletableFuture<>();
         final OperationContext context = streamMetadataStore.createContext(request.getScope(), request.getStream());
 
-        // TODO: implement update stream task (issue 1738)
-        // 1. check precondition --> update version
-        // 2. set state updating
-        // 3. update
-        // 4. set state to active
+        // when update stream event is picked, update task is performed.
```

Comment:
Some private methods would help make this clearer.

Reply:
done